packages/react-components/react-skeleton/SPEC.md

```diff
@@ -0,0 +1,107 @@
+# @fluentui/react-skeleton Spec
+
+## Background
+
+The `Skeleton` component is a temporary animation placeholder for when a service call takes time to return data and we don't want to block rendering the rest of the UI.
+
+## Prior Art
+
+### Open UI
+
+| Library    | Component Name | Spec Link                                                                       | Notes                                                                                                                                                                  |
+| ---------- | -------------- | ------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Ant Design | Skeleton       | [Skeleton](https://ant.design/components/skeleton/)                             | Has an `avatar`, `button`, `list`, and `togglebutton`. Has animated and non-animated versions, and shape variants for the component skeletons(rounded or rectangular). |
+| Fast       | Skeleton       | [Skeleton](https://explore.fast.design/components/fast-progress)                | Has three components: `Avatar`, `List`, and `Button`. No shape variants.                                                                                               |
+| UI Fabric  | Shimmer        | [Shimmer](https://developer.microsoft.com/en-us/fluentui#/controls/web/shimmer) | Shimmer has a way to pass in custom elements as well as the defaults. There is also a prop to change the colors of the Skeleton components.                            |
```

Comment:
FYI https://react.semantic-ui.com/elements/placeholder/

Reply:
What do you see as the advantages of this API compared with the existing Fluent one?

Considering that we've avoided layout components like `Stack` in Fluent v9 it seems we should also avoid adding components like `Skeleton.Header` as it appears to just be for layout (correct me if I'm wrong here though!)

Reply:
> What do you see as the advantages of this API compared with the existing Fluent one?

I don't see API examples in the spec currently (or in #25452), so it's hard to say 😉 

> Considering that we've avoided layout components like Stack in Fluent v9 it seems we should also avoid adding components like Skeleton.Header as it appears to just be for layout (correct me if I'm wrong here though!)

Yes, correct. It applies special kind of styling.

-----

My note there was to consider different approaches. For example, in v8 its:

```tsx
<ShimmerElementsGroup
  shimmerElements={[
    { type: ShimmerElementType.line, width: 40, height: 40 },
    { type: ShimmerElementType.gap, width: 10, height: 40 },
  ]}
/>
```

Which does not seem to be v9 way at all 🐱 That's why I mentioned SUIR as it something like this might be more scalable and customisable:

```tsx
<Skeleton>
  <SkeletonLine />
  <SkeletonGap />
</Skeleton>
```

However, the spec does not contain API examples yet so I don't know in which direction we are going currently.